solr/core/src/java/org/apache/solr/search/CaffeineCache.java

```diff
@@ -98,6 +98,13 @@ public class CaffeineCache<K, V> extends SolrCacheBase
   private int maxIdleTimeSec;
   private boolean cleanupThread;
   private boolean async;
+  private final ThreadLocal<boolean[]> detectRecursion =
+      new ThreadLocal<>() {
+        @Override
+        protected boolean[] initialValue() {
```

Comment:
Why a boolean array; why not a Boolean?  We can easily change the ThreadLocal value; it's not immutable.
Also, it's less code to use ThreadLocal.[withInitial](https://www.tabnine.com/code/java/methods/java.lang.ThreadLocal/withInitial)